contributors/design-proposals/storage/csi-migration.md

```diff
@@ -841,6 +841,43 @@ When the above conditions are not met, the PV Controller will use FindProvisiona
 to determine the in-tree plugin that can be used for provisioning a volume (as is
 the case today).
 
+Update: 1/13/2020
+
+In Beta we discovered issue: https://github.com/kubernetes/kubernetes/issues/79043
+
+In order to resolve this the design needs to be modified. When migration is "on"
+the Persistent Volume Controller will still set the Storage Provisioner
+annotation (volume.beta.kubernetes.io/storage-provisioner) with the name of the
+migrated CSI driver. However, it will also set an additional annotation
+`volume.beta.kubernetes.io/migrated-to` to the CSI Driver name. Additionally,
+the PV Controller will be modified so that it lists all PV's and PVC's on
+startup and checks the `volume.beta.kubernetes.io/storage-provisioner` and
+`pv.kubernetes.io/provisioned-by` annotations respectively to anneal the
+required `volume.beta.kubernetes.io/migrated-to` annotation. An example is a PV
+with `volume.beta.kubernetes.io/storage-provisioner=kubernetes.io/gce-pd`, when
+the PV controller starts up with `CSIMigrationGCE=true` the controller will
+additionally annotate the PV with
+`volume.beta.kubernetes.io/migrated-to=pd.csi.storage.gke.io`. The PV controller
+will also remove `migrated-to` annotations on PV/PVCs with migration OFF to
+support rollback scenarios.
```

Comment:
Please mention that there is race between PV controller removing the annotation and external-provisioner deleting the volume. We depend heavily on idempotency both in CSI drivers and in-tree volume plugins. Deleting a volume already deleted or being deleted by something else should return "success". Especially the "being deleted" part is new.